relay-server/src/envelope/item.rs

```diff
@@ -191,6 +192,35 @@ impl Item {
         }
     }
 
+    /// Returns the number of spans in `event.spans`.
+    pub fn span_count(&self) -> usize {
+        self.headers.span_count.unwrap_or(0)
+    }
+
+    #[cfg(test)]
+    pub fn set_span_count(&mut self, value: usize) {
+        self.headers.span_count = Some(value);
+    }
+
+    /// Sets the `span_count` item header by shallow parsing the event.
+    ///
+    /// Returns the recomputed count.
+    pub fn refresh_span_count(&mut self) -> usize {
+        let count = self.parse_span_count();
+        self.headers.span_count = count;
+        count.unwrap_or(0)
+    }
+
+    /// Returns the span_count header, and computes it if it is not yet set.
```

Comment:
```suggestion
    /// Returns the `span_count` header, and computes it, if it has not yet been set.
```

Not 100% sure on that grammar.